sklearn/neighbors/_nearest_centroid.py

```diff
@@ -255,7 +307,95 @@ def predict(self, X):
         """
         check_is_fitted(self)
 
-        X = self._validate_data(X, accept_sparse="csr", reset=False)
-        return self.classes_[
-            pairwise_distances_argmin(X, self.centroids_, metric=self.metric)
-        ]
+        d = self._decision_function(X)
```

Comment:
I don't like that pattern here. `pairwise_distances_argmin` is highly efficient because it will never materialize in memory the discriminant scores but take the `argmin` on the chunk. I think that we should be able to call `pairwise_distances_argmin` in predict to benefit from this work.

Reply:
Maybe I can call pairwise_distances_argmin in the __decision_function_?